pkgs/by-name/dr/draupnir/package.nix

```diff
@@ -0,0 +1,96 @@
+{ lib
+, fetchFromGitHub
+, makeWrapper
+, nodejs
+, matrix-sdk-crypto-nodejs
+, mkYarnPackage
+, fetchYarnDeps
+, nixosTests
+}:
+
+# docs: https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/javascript.section.md#yarn2nix-javascript-yarn2nix
+let
+  hashesFile = builtins.fromJSON (builtins.readFile ./hashes.json);
+in
+  mkYarnPackage rec {
+    pname = "draupnir";
+    version = "2.0.0-beta.4";
+    src = fetchFromGitHub {
+      owner = "the-draupnir-project";
+      repo = "Draupnir";
+      rev = "v${version}";
+      hash = "sha256-ZbAst3XaTdJxCPsgPc8cAqN0lo6vOvFphqBDjduQ/to=";
+    };
+
+    nativeBuildInputs = [
+      makeWrapper
+    ];
+
+    offlineCache = fetchYarnDeps {
+      name = "${pname}-yarn-offline-cache";
+      yarnLock = src + "/yarn.lock";
+      hash = hashesFile.yarn_offline_cache_hash;
+    };
+
+    yarnLock = src + "/yarn.lock";
+    packageJSON = ./package.json;
```

Comment:
I don't think we need these? I can build the package without setting these.

Reply:
This confused me as well to figure out, the package builds fine without these but the nixpkgs CI fails due to importing the package.json from source. I'm not sure what the yarnLock there is for though as it seems to be duplicated from offlineCache.yarnLock

Reply:
I see, apparently this causes IFD. Disregard this then ^^

Reply:
Unresolved this conversation, does anyone think it would be possible to address this in favor of https://github.com/NixOS/nixpkgs/issues/320528 and https://github.com/NixOS/nixpkgs/issues/327064?

Reply:
No, that would be scope creep at this point.